### PR TITLE
AE-2282: (AED-750) Adding all events to monitor Multisig contract

### PR DESCRIPTION
**Pull request by arbitrary-echo**
This work updates the Compound Multisig Monitor Bot to check for all possible events emitted by the Compound Community Multisig smart contract.

### CODE REVIEW

File: multisig-transactions-monitor/src/agent.spec.js [hunk 1]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+        },
+      });
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the ChangedMasterCopy event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:274]:
```suggestion
      // use ChangedMasterCopy event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 2]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the ChangedThreshold event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      const thresholdValue = ethers.BigNumber.from(0);
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:307]:
```suggestion
      // use ChangedThreshold event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 3]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+        },
+      });
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the DisabledModule event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:338]:
```suggestion
      // use DisabledModule event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 4]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+        },
+      });
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the EnabledModule event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:369]:
```suggestion
      // use EnabledModule event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 5]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the ExecutionFailure event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      const paymentValue = ethers.BigNumber.from(0);
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:402]:
```suggestion
      // use ExecutionFailure event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 6]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+        },
+      });
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the ExecutionFromModuleFailure event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:434]:
```suggestion
      // use ExecutionFromModuleFailure event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 7]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+        },
+      });
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the ExecutionFromModuleSuccess event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:465]:
```suggestion
      // use ExecutionFromModuleSuccess event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 8]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the ExecutionSuccess event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      const paymentValue = ethers.BigNumber.from(0);
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:498]:
```suggestion
      // use ExecutionSuccess event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 9]
```diff
@@ -221,16 +219,317 @@ describe('monitor multisig contract transactions', () => {
...
+        },
+      });
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the RemovedOwner event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:530]:
```suggestion
      // use RemovedOwner event to test
```

File: multisig-transactions-monitor/src/agent.spec.js [hunk 10]
```diff
@@ -240,16 +539,47 @@ describe('monitor multisig contract transactions', () => {
...
+        },
+      });
+
+      expect(findings).toStrictEqual([expectedFindings]);
+    });
+
+    it('returns findings if multisig was involved in a transaction and the SignMsg event was emitted', async () => {
+      mockTxEvent.addresses[multisigAddress] = true;
+
+      // use AddedOwner event to test
```
Comment (darian-ae) [inline multisig-transactions-monitor/src/agent.spec.js:561]:
```suggestion
      // use SignMsg event to test
```

**Review by darian-ae — COMMENTED** [review]
Looks good, just added a few nits to the testing spec. Another thing to consider is that we might want to update the `README` to reflect the additional alerts that were added

**Review by teasipper-ae — APPROVED** [review]

**Review by arbitrary-nomad — APPROVED** [review]
Looks good and all tests run

File: defender/forta_multi_sig/autotask-1/index.js [hunk 11]
```diff
@@ -174,10 +174,50 @@ exports.handler = async function (autotaskEvent) {
       ({ owner } = metadata);
       message = `🆕 **Added Owner** ${owner} to Community Multi-Sig${protocolVersionString}`;
       break;
+    case 'AE-COMP-MULTISIG-APPROVED-HASH-ALERT':
```
Comment (arbitrary-nomad) [inline defender/forta_multi_sig/autotask-1/index.js:177]: fix the cspell errors, add this:
    `megalinter_file_names_cspell.txt`
to `ignore paths` in `compound-monitoring/.github/cspell.json`

File: multisig-transactions-monitor/src/agent.spec.js (hunk 1 above)
Comment (arbitrary-echo) [reply multisig-transactions-monitor/src/agent.spec.js:274]: Fixed.

File: multisig-transactions-monitor/src/agent.spec.js (hunk 2 above)
Comment (arbitrary-echo) [reply multisig-transactions-monitor/src/agent.spec.js:307]: Fixed.

File: multisig-transactions-monitor/src/agent.spec.js (hunk 3 above)
Comment (arbitrary-echo) [reply multisig-transactions-monitor/src/agent.spec.js:338]: Fixed.

File: multisig-transactions-monitor/src/agent.spec.js (hunk 4 above)
Comment (arbitrary-echo) [reply multisig-transactions-monitor/src/agent.spec.js:369]: Fixed.

File: multisig-transactions-monitor/src/agent.spec.js (hunk 7 above)
Comment (arbitrary-echo) [reply multisig-transactions-monitor/src/agent.spec.js:465]: Fixed.

File: multisig-transactions-monitor/src/agent.spec.js (hunk 9 above)
Comment (arbitrary-echo) [reply multisig-transactions-monitor/src/agent.spec.js:530]: Fixed.

File: multisig-transactions-monitor/src/agent.spec.js (hunk 10 above)
Comment (arbitrary-echo) [reply multisig-transactions-monitor/src/agent.spec.js:561]: Fixed.

**Review by darian-ae — APPROVED** [review]

File: defender/forta_multi_sig/autotask-1/index.js (hunk 11 above)
Comment (arbitrary-echo) [reply defender/forta_multi_sig/autotask-1/index.js:177]: Done.

**Pull request merged**